lit_parrot/utils.py

```diff
@@ -425,3 +425,23 @@ def save(self) -> None:
     logger.experiment.save = MethodType(save, logger.experiment)
 
     return logger
+
+
+def chunked_cross_entropy(logits: torch.Tensor, targets: torch.Tensor, chunk_size: int = 128) -> torch.Tensor:
```

Comment:
As described in the comment below, at the beginning of `backward`, we get very sharp memory spikes based on the `max_seq_length`

Notice the super sharp peaks (with data samples padded to max_seq_length="1307"):

Before, no chunking (1.2 GB max allocated):
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/3b80bca3-0429-408a-b848-bf4b426add32)

After, chunked to 128 (1 GB max allocated):
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/5b5ca578-da3c-4cde-a50e-0588b23e9beb)

128 seemed to work well in the small trials I ran, but there might be more optimal sizes for different models or sequence lengths